File: 07. Binary-Search/python/binarySearch.py
# Problem Description 

# Implement Binary Search

def binarySearch(arr, target):

    left = 0
    right = len(arr)-1

    while(left <= right):

        mid = (left + right) // 2

        if(arr[mid] == target):
            return True
        elif(arr[mid] < target):
            right = mid - 1
        elif(arr[mid] > target):
            left = mid + 1
    
    return False

arr = list(map(int, input("Enter array elements: ").split()))
target = int(input("Enter target: "))

print(binarySearch(arr, target));

# Time Complexity  : O(log n)
# Space Complexity : O(1)

File: 07. Binary-Search/javascript/binarySearch.js
// Problem Description 

// Implement Binary Search

function binarySearch(arr, target) {

    let left = 0;
    let right = arr.length-1;

    while(left <= right) {

        let mid = Math.floor((left + right) / 2);

        if(arr[mid] ===  target) {
            return true
        }
        else if(arr[mid] > target) {
            right = mid - 1;
        }
        else if(arr[mid] < target) {
            left = mid + 1;
        }
    }
    return false;

}

binarySearch([1,2,3,4,5], 1);

/*
    Time Complexity  : O(log n)
    Space Complexity : O(1)
*/
 